x-pack/plugins/serverless/server/plugin.ts

```diff
@@ -36,6 +37,11 @@ const typeToIdMap: Record<ProjectType, string> = {
 export class ServerlessPlugin implements Plugin<ServerlessPluginSetup, ServerlessPluginStart> {
   private readonly config: ServerlessConfig;
 
+  private setupProjectSettings(core: CoreSetup, keys: string[]): void {
+    const settings = [...ALL_COMMON_SETTINGS].concat(keys);
+    core.uiSettings.setAllowlist(settings);
+  }
```

Comment:
- if the plugin receives multiple calls to `setupProjectSettings`, then only the value from the last one will effectively be used.
- if the plugin receives **no** call to `setupProjectSettings`, then `core.uiSettings.setAllowlist` will never be called, meaning that all settings will be enabled / visible.

Are we fine with this? I know that "in theory", exactly one call to `setupProjectSettings` will be done for any project type in serverless, but how much do we want to rely blindly on this "theory" (e.g we add a 4rth project type later, by default all settings will be enabled and visible)?

Reply:
cc @jloleysens FYI

Yeah, that's not the first time we're facing problems with this script that forces all plugins to be enabled at the same time (which is something that was fine when we designed the thing but no longer with the "project type only" plugins.

I don't have any great suggestions right now, so I would just disable this check for now to avoid blocking the PR, and try to address it as a follow up.